spec/controllers/api/admin/updates_controller_spec.rb

```diff
@@ -42,4 +42,42 @@
       end
     end
   end
+
+  describe 'GET #show' do
+    context 'when records are present' do
+      subject(:do_request) { get :show, params: { id: update.to_param, format: :json } }
+
+      let(:pattern) do
+        {
+          data: {
+            id: String,
+            type: 'tariff_update',
+            attributes: {
+              update_type: 'TariffSynchronizer::CdsUpdate',
+              state: String,
+              filename: String,
+              created_at: String,
+            }.ignore_extra_keys!,
+          }.ignore_extra_keys!,
+        }
+      end
+
+      let(:update) { create(:cds_update) }
+
+      it { expect(do_request.body).to match_json_expression pattern }
+    end
+
+    context 'when records are not present' do
+      subject(:do_request) { get :show, params: { id: 'foo', format: :json } }
+
+      let(:pattern) do
+        {
+          error: 'not found',
+          url: 'http://test.host/admin/updates/foo',
+        }
+      end
+
+      it { expect(do_request.body).to match_json_expression pattern }
+    end
+  end
```

Comment:
I thought we more in favour of request spec, rather then controller.

Reply:
I can migrate this separately